### PR TITLE
2024 Q1 articat housekeeping and upgrades

### PR DESCRIPTION
**Pull request by dhimmel**

### CODE REVIEW

File: articat/config.py [hunk 1]
```diff
@@ -162,7 +162,8 @@ def fs_tmp_prefix(self) -> str:
             r = tempfile.mkdtemp(suffix="artifact_temp_dir_")
             self._set_option(self._config, "fs", "tmp_prefix", r)
             warnings.warn(
-                f"FSArtifact temp directory not configured, assuming local mode, using temp directory: {r}"
+                f"FSArtifact temp directory not configured, assuming local mode, using temp directory: {r}",
+                stacklevel=1,
```
Comment (danielhfrank) [inline articat/config.py:166]: 🟢 what do these `stacklevel` args do? Is there anything about this that we should keep in mind or bring with us to the facets repo?

**Review by danielhfrank — APPROVED** [review]
Looks good to me. Guessing the formatting changes are the result of running `ruff` on everything?

File: articat/config.py (hunk 1 above)
Comment (dhimmel) [reply articat/config.py:166]: `stacklevel=1` is the default, so stuck with that. Needed to add to quiet [no-explicit-stacklevel (B028)](https://docs.astral.sh/ruff/rules/no-explicit-stacklevel/). I didn't think more verbose warnings with an additional level of stack trace were necessary here, but also did not really think about it.

**Comment by dhimmel** [comment]
> Guessing the formatting changes are the result of running ruff on everything?

Yeah, although many of them could be from updating black before removing it, so I cannot say the balance of black versus ruff induced changes.

**Pull request merged**